File: 3411-maximum-subarray-with-equal-products/3411-maximum-subarray-with-equal-products.java
class Solution {
    public int maxLength(int[] nums) {
        int maxLength = 0;
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int prod = 1;
            int gcdVal = nums[i];
            int lcmVal = nums[i];
            for (int j = i; j < n; j++) {
                prod *= nums[j];
                gcdVal = gcd(gcdVal, nums[j]);
                lcmVal = lcm(lcmVal, nums[j]);
                if (prod == (long) lcmVal * gcdVal) {
                    maxLength = Math.max(maxLength, j - i + 1);
                }
            }
        }

        return maxLength;
    }

    private int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }

    private int lcm(int a, int b) {
        return a * (b / gcd(a, b));
    }
}

File: 3411-maximum-subarray-with-equal-products/3411-maximum-subarray-with-equal-products.py
class Solution:
    def maxLength(self, nums: List[int]) -> int:
        maxLength = 0
        n = len(nums)
        

        for i in range(n):
            prod = 1
            gcdVal = nums[i]
            lcmVal = nums[i]
            j = 1
            prod *= nums[j]
            gcdVal = gcd(gcdVal, nums[j])
            lcmVal = lcm(lcmVal, nums[j])
            if prod == lcmVal * gcdVal:
                maxLength = max(maxLength, j - i + 1)
        return maxLength

